[netrify] failed to  - too lower golang version.

**What happened**:

- your gateway-api netrify build failed. 
- https://github.com/kubernetes-sigs/gateway-api/pull/2647#issuecomment-1840978106

- relates
  - #2642

``
/opt/buildhome/.gimme_cache/gopath/pkg/mod/github.com/go-logr/logr@v1.3.0/logr.go:562:15: undefined: any
9:24:00 PM: note: module requires Go 1.18
9:24:00 PM: make: *** [Makefile:127: docs] Error 2
``

it has go 1.14.4, but required 1.18.

``
9:23:44 PM: Started restoring cached go cache
9:23:45 PM: Finished restoring cached go cache
9:23:45 PM: Installing Go version 1.14.4 (requested 1.14.4)
9:23:49 PM: go version go1.14.4 linux/amd64
``

**What you expected to happen**:

- installed go 1.20 or newer.

**How to reproduce it (as minimally and precisely as possible)**:

bump k8s.io/klog/v2 to v2.110.1

**Anything else we need to know?**:

you can set set `GO_VERSION` [Environment variables](https://app.netlify.com/sites/stellar-flan-49fb62/configuration/env#environment-variables) 
and [Clear cache and deploy site](https://answers.netlify.com/t/cache-issue-on-netlify/8238)

## netrify
https://docs.netlify.com/configure-builds/manage-dependencies/#go
https://docs.netlify.com/configure-builds/environment-variables/

--- a/netlify.toml
+++ b/netlify.toml
@@ -4,7 +4,7 @@
     publish = "site"
 [build.environment]
     PYTHON_VERSION = "3.8"
-    GO_VERSION = "1.21.4"
+    GO_VERSION = "1.21"
 
 # Standard Netlify redirects
 [[redirects]]

--- a/netlify.toml
+++ b/netlify.toml
@@ -4,7 +4,7 @@
     publish = "site"
 [build.environment]
     PYTHON_VERSION = "3.8"
-    GO_VERSION = "1.21"
+    GO_VERSION = "1.21.4"
 
 # Standard Netlify redirects
 [[redirects]]

--- a/netlify.toml
+++ b/netlify.toml
@@ -2,7 +2,9 @@
 [build]
     command = "make docs"
     publish = "site"
-    environment = { PYTHON_VERSION = "3.8" }
+[build.environment]
+    PYTHON_VERSION = "3.8"
+    GO_VERSION = "1.21.4"
 
 # Standard Netlify redirects
 [[redirects]]

